src/libstore/derivations.hh

```diff
@@ -13,10 +13,20 @@ namespace nix {
 
 /* Abstract syntax of derivations. */
 
-struct DerivationOutput
+struct DerivationOutputIntensional
```

Comment:
Should be `DerivationOutputExtensional` (though perhaps we might want to drop the "extensional" vs "intensional" jargon)

Reply:
oh right

Reply:
do we have a better name than intensional / extensional?

Reply:
I tend to use content-addressed/input-addressed. That's more verbose but doesn't require 10s of brain power to remember which one is which

Reply:
"input addressed" is great!